frb_codegen/src/generator/rust/ty_enum.rs

```diff
@@ -37,7 +37,16 @@ impl TypeRustGeneratorTrait for TypeEnumRefGenerator<'_> {
                                 };
 
                                 if !target.is_wasm() {
-                                    format!("{field_} ans.{field_name}.wire2api()")
+                                    let shared_mod_name: Option<String> =
+                                        self.get_shared_module_of_a_type(&field.ty);
+                                    if !self.context.config.shared && shared_mod_name.is_some() {
+                                        format!(
+                                            "{field_} {}::Wire2Api::wire2api(ans.{field_name})",
+                                            shared_mod_name.unwrap()
+                                        )
+                                    } else {
+                                        format!("{field_} ans.{field_name}.wire2api()")
+                                    }
```

Comment:
nit: seems that this `Wire2api::wire2api` pattern is seen twice (here, and another place), maybe we can extract it.